src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,47 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def alert_rule_data_requires_async_lookup(organization, user, data, write_input_channel_id=False):
+    try:
+        from sentry.incidents.endpoints.serializers import AlertRuleTriggerActionSerializer
+
+        for trigger in data["triggers"]:
+            for action in trigger["actions"]:
+                a_s = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "access": SystemAccess(),
+                        "user": user,
+                        "use_async_lookup": True,
+                    },
+                    data=action,
+                )
+                if a_s.is_valid():
+                    if a_s.validated_data["type"].value in AlertRuleTriggerAction.INTEGRATION_TYPES:
+                        if (
+                            a_s.validated_data["type"].value
+                            == AlertRuleTriggerAction.Type.SLACK.value
+                            and not a_s.validated_data["input_channel_id"]
+                        ):
+                            if not write_input_channel_id:
+                                return True
+                            else:
+                                (
+                                    target_identifier,
+                                    target_display,
+                                ) = get_target_identifier_display_for_integration(
+                                    a_s.validated_data["type"].value,
+                                    a_s.validated_data["target_identifier"],
+                                    organization,
+                                    a_s.validated_data["integration"].id,
+                                    use_async_lookup=True,
+                                    input_channel_id=None,
+                                )
+                            if target_identifier:
+                                action["input_channel_id"] = target_identifier
+    except Exception:
```

Comment:
@iProgramStuff I think a blanket catch of any exception is dangerous here, we might be masking a real bug

Reply:
The problem I had here was that we are looping through raw data submitted through the API.

I think I need to add more checks/validation here on the data, or maybe catch specific index errors accessing triggers/actions. I'll work on improving this.

Reply:
Changed it to catch a KeyError (from accessing triggers/actions when they are not there). I'll let other errors happen.